pymbar/mbar.py

```diff
@@ -307,7 +305,7 @@ def __init__(self, u_kn, N_k, maximum_iterations=10000, relative_tolerance=1.0e-
             self._initializeFreeEnergies(verbose, method=initialize)
 
             if self.verbose:
-                logger.info("Initial dimensionless free energies with method {}".format(initialize))
+                logger.info("Initial dimensionless free energies with method {:s}",format(initialize))
```

Comment:
This should be `.format` (notice the comma). Also `{:s}` is totally equivalent to `{}`, so I don't think we should go this way. I think what `pylint` was trying to say is that we should use old-style formatting (`%` based) for the logging calls!

I hadn't realized this but some there are _some_ good points which might or might not be irrelevant to us (taken from [here](https://www.reddit.com/r/Python/comments/387oog/when_will_logging_use_newstyle_string_formatting/crsz00v/)):

- Since the arguments are separate from the template, the `logging` module can choose when to apply the string interpolation/replacement saving some CPU cycles. This is, the message will only be formatted if the logging level is sufficient.
- Having the template available means that logging aggregators (like `sentry`) can categorize log messages way better.

So in other words, `pylint` is trying to say that we should write it this way:


```suggestion
                logger.info("Initial dimensionless free energies with method %s", initialize)
```

Reply:
Oh, there's a compromise: http://plumberjack.blogspot.com/2010/10/supporting-alternative-formatting.html

Reply:
Oh, apparently, there's a compromise. We can use new-style formatting with `logging` without losing the performance and aggregation abilities following [these instructions](https://stackoverflow.com/a/24683360).